### ISSUE TITLE
OMEMO-enabled MUC loses offline messages when sent by pidgin

### ISSUE THREAD

**Issue by t2d** [gkdr/lurch#101]
We tried a chat with 3 participant:
* Pidgin
* OMEMO
* ChatSecure

Without OMEMO everything works as expected. Every client receives offline messages of any other client.

If we enable OMEMO on all devices, Gajim and ChatSecure cannot receive offline messages sent by Pidgin.  
Offline messages sent by any other client can be received by all clients.

**Comment by t2d** [comment]
Yes, all clients we're always present during the history replay.

**Comment by t2d** [comment]
I believe this is what the ChatSecure message looks like (taken with Gajim during live muc):
```
<message xmlns="jabber:client" id="9B24BEA2-8C17-4B56-A4B0-36C8C23F67DB" type="groupchat" to="client@example.org/gajim.X0M4OAP6" from="test_omemo@conference.example.org/ChatSecure">
  <store xmlns="urn:xmpp:hints" />
  <encrypted xmlns="eu.siacs.conversations.axolotl">
    <header sid="1236947956">
      <key rid="876859048">MwohBX2oXTW3TMP24Rv+L7fZKZC0d0YpAGI76V0SE7u8iwIbEAAYACIwaL1yb2t5Li5oNnW6vHrY08eISu/uzfhDg2nwJhbBDaDwNyEhCb5C0/LvcdewlQV/ZoyVrqXsM0I=</key>
      <key rid="546442800">MwohBfe6/vlDjm3KZOTUt0NDzCRyOhvHB0gp6JOF/PQ+bJkAEAQYACIwR/ZpPpakEqCRnorb4Uc7HIwAYtLWpV1dx+aHfT9zqWsyCvRgI4HYPD5LecWHewg6RujbME2jgbQ=</key>
      <key rid="669082632">MwohBe0157ImPjjY/cNMD6ekzO6+kNqcafDM/YPrFHGuRYhyEAAYACIwePx3SLWPK/hk/h5EgedWZzgWnhpC4IFbdax1schNFoc0eOwkkSKBfNDI9IBFPwT1ZhS/FilVBoE=</key>
      <iv>JGTzQ8Pc1WbXdRJTluppAA==</iv>
</header>
    <payload>mQECHw==</payload>
</encrypted>
  <origin-id id="9B24BEA2-8C17-4B56-A4B0-36C8C23F67DB" xmlns="urn:xmpp:sid:0" />
</message>
```

**Comment by t2d** [comment]
To make MUC use MAM prosody uses [mod_mam_muc](https://modules.prosody.im/mod_mam_muc.html)

**Comment by gkdr** [comment]
Oh, okay, that probably explains it. I was only looking at the README.

So I suppose the Gajim messages do have a `<body>`, so they're part of the regular MUC history and that's why Pidgin can decrypt them, but only if the sender is online at the time the history replay happens.

I don't understand why messages by Pidgin are ignored then though, since I do add the same hint as ChatSecure. Maybe because Pidgin can't do MAM and doesn't advertise it, but I'm not sure if that makes sense because it doesn't matter to recipients whether the sender can do MAM or not. Or maybe I should put the hint first just like ChatSecure. If you want, you can play around with the XMPP Console a bit and see if that helps. MAM support would be an entirely different matter though, there's so many moving parts I can't seem to get this somewhat stable even without it.
